### ISSUE TITLE
NFS credentials

### ISSUE THREAD

**Issue by santimar** [kubernetes-csi/csi-driver-nfs#525]
**Is your feature request related to a problem?/Why is this needed**
Right now it seems that this driver can only mount NFS volumes if server allow anonymous access.
2 years ago, on https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/issues/136#issuecomment-907710848 was said that this feature could have been implemented here  but I wasn't able to find anything related.

**Describe the solution you'd like in detail**
A way to set some kind of credentials to perform authentication (eg Kerberos, sys) 

**Additional context**
none

**Comment by viceice** [comment]
need this too. on ionos the shared storage uses kerberos 😕

**Comment by BloodyIron** [comment]
Yeah can we please get authentication options for NFS for this CSI driver? It's very insecure to expect NFS interfacing without authentication over a network. I'm not even sure where encryption fits into this picture, but I'm more concerned now that I start thinking about that too.

Authentication and security aspects for NFS have been around for about a decade, really need this to be added please!

**Comment by BloodyIron** [comment]
Yeah definitely still interested.

**Comment by BloodyIron** [comment]
So what's going on here then? Hmmm?

**Comment by BloodyIron** [comment]
STILL INTERESTED CAN WE PLEASE GET THIS? ugh like at least added to some sort of kanban board or tracking list?

**Comment by andyzhangx** [comment]
**Current status:** The NFS CSI driver only supports `sec=sys` (UID/GID-based access control) and does not have built-in Kerberos (`sec=krb5/krb5i/krb5p`) support.

**Why this is harder than SMB Kerberos:**

The [SMB CSI driver](https://github.com/kubernetes-csi/csi-driver-smb) already supports Kerberos by accepting a base64-encoded ticket cache (ccache) via CSI Secrets and writing it to the node filesystem before mounting. This works because `mount.cifs` can directly consume ccache files.

NFS Kerberos is fundamentally different — it relies on the `rpc.gssd` daemon running on the node to handle RPCSEC_GSS authentication. The driver cannot simply write a ccache file; it needs:
- A valid keytab on each node
- `rpc.gssd` daemon running on the host
- Proper principal mapping for each pod identity
- Ticket renewal and keytab rotation lifecycle management

**Workarounds available today:**

1. **Use `mountOptions` with `sec=krb5`** — If your nodes already have Kerberos configured (keytab + gssd), you can pass `sec=krb5` in `StorageClass.mountOptions` and the NFS CSI driver will pass it through to the mount command.

2. **Use a DaemonSet to configure Kerberos on nodes** — Deploy a privileged DaemonSet that sets up keytabs and ensures `rpc.gssd` is running on each node. Once the host Kerberos environment is ready, NFS mounts with `sec=krb5` will work transparently.

3. **Use SMB/CIFS instead** — If your storage supports both NFS and SMB, consider using the [SMB CSI driver](https://github.com/kubernetes-csi/csi-driver-smb) which has first-class Kerberos support.

**Comment by BloodyIron** [comment]
SMB can't do file locking like NFS can, there are plenty of implementations where SMB just won't let an application "do its job" where NFS will.

Certainly appreciate the nuanced details on how we can make this achievable, but I'm already using SMB CSI Driver stuff in areas in addition to NFS, and I have scenarios where I _cannot_ use SMB I have to use NFS instead (I'm not using iSCSI for long-winded data efficiency reasons).

Also, the protocol is SMB, not CIFS, it hasn't been CIFS since 1999 no matter how many NetApp "Engineers" say otherwise. It's been SMB this whole time, let's use the accurate term if need be, SMB, not CIFS. It's not CIFS as CIFS is only SMB v1.0 and you don't want that.

**Comment by BloodyIron** [comment]
NFS credentials is something that really needs to be added, so please don't close this.
